integration-tests/bats/backup.bats

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env bats
+load $BATS_TEST_DIRNAME/helper/common.bash
+
+setup() {
+    setup_common
+    TMPDIRS=$(pwd)/tmpdirs
+    mkdir -p $TMPDIRS/{rem1,bac1,repo1}
+
+    # repo1 -> rem1 -> repo2
+    cd $TMPDIRS/repo1
+    dolt init
+    dolt tag v1
+    dolt sql -q "create table t1 (a int)"
+    dolt commit -am "cm"
+    dolt branch feature
+    dolt remote add origin file://../rem1
+    dolt push origin main
+    cd $TMPDIRS
+}
+
+teardown() {
+    teardown_common
+    rm -rf $TMPDIRS
+    cd $BATS_TMPDIR
+}
+
+@test "backup: add named backup" {
+    cd repo1
+    dolt backup add bac1 file://../bac1
+    run dolt backup -v
+    [ "$status" -eq 0 ]
+    [ "${#lines[@]}" -eq 1 ]
+    [[ "$output" =~ "bac1" ]] || false
+}
+
+@test "backup: remove named backup" {
+    cd repo1
+    dolt backup add bac1 file://../bac1
+    run dolt backup -v
+    [ "$status" -eq 0 ]
+    [ "${#lines[@]}" -eq 1 ]
+    [[ "$output" =~ "bac1" ]] || false
+
+    dolt backup remove bac1
+
+    run dolt backup -v
+    [ "$status" -eq 0 ]
+    [ "${#lines[@]}" -eq 0 ]
+    [[ ! "$output" =~ "bac1" ]] || false
+}
+
+@test "backup: rm named backup" {
+    cd repo1
+    dolt backup add bac1 file://../bac1
+    run dolt backup -v
+    [ "$status" -eq 0 ]
+    [ "${#lines[@]}" -eq 1 ]
+    [[ "$output" =~ "bac1" ]] || false
+
+    dolt backup rm bac1
+
+    run dolt backup -v
+    [ "$status" -eq 0 ]
+    [ "${#lines[@]}" -eq 0 ]
+    [[ ! "$output" =~ "bac1" ]] || false
+}
+
+@test "backup: sync master to backup" {
```

Comment:
Most of these tests below this point are actually tests of restore as well as sync, and could probably be combined into a single test with more assertions

Reply:
i wanted to use `noms ds .dolt/noms` to separate sync and restore tests, but was having issues with the noms command in windows tests. i think i prefer a separation of tests by dataset type to clarify what's broken if someone does a bad refactor?